### PR TITLE
Add recipe for duckdb

### PR DESCRIPTION
**Pull request by xhochy**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [ ] Package does not vend other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by xhochy** [comment]
@conda-forge/staged-recipes This is ready for review.

Once some upstream fixes were done (see https://github.com/cwida/duckdb/issues/225), I will also submit packages for R and C++. I would call them `duckdb-cpp` and `r-duckdb`. Thus is `duckdb` as on pypi the right choice for this package?

File: recipes/duckdb/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+  host:
+    - python
+    - pip
+    - numpy 1.16
+    - pandas
+  run:
+    - python
+    - numpy
```
Comment (chrisburr) [inline recipes/duckdb/meta.yaml]: If you're using the numpy C API then you need this:
```suggestion
    - {{ pin_compatible('numpy') }}
```
See: https://conda-forge.org/docs/maintainer/knowledge_base.html#building-against-numpy

**Comment by chrisburr** [comment]
For the naming, I don't think conda-forge has any guidelines for this. It looks like `duckdb` is the main package and the others are bindings built on top of it so I'd be tempted to call them `duckdb`, `python-duckdb` and `r-duckdb`. But, as there aren't any guidelines, it's up to you.

I'm happy for you to self-merge this either way.

File: recipes/python-duckdb/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+{% set name = "duckdb" %}
+{% set version = "0.1.0" %}
+
+package:
+  name: python-{{ name|lower }}
```
Comment (xhochy) [inline recipes/python-duckdb/meta.yaml]: @chrisburr I changed the name like, do you think this is ok or do we need the `name` variable to match the actual package name?

Comment (chrisburr) [reply recipes/python-duckdb/meta.yaml]: The jinja2 variable doesn't matter so it's fine like this.

Have you tried installing with pip like the linter now suggests?

Comment (xhochy) [reply recipes/python-duckdb/meta.yaml]: Yes, and I've raised https://github.com/cwida/duckdb/issues/227 upstream.

Comment (chrisburr) [reply recipes/python-duckdb/meta.yaml]: Great, LGTM 🎉

**Pull request merged**